### PR TITLE
Choose sequence card type

### PR DESCRIPTION
**Pull request by zenby**
There is very rough implementation of `choose_sequence` card. If you have any suggestions it would be helpful to share them with me.

TODO:
- [x] - remove `test.js` file after development;
- [x] - remove `index.html` file after development;
- [x] - remember and show answer after click to `check` button;
- [x] - fix test to have valid card when run `test:card-types`;
- [x] - add README.md;
- [x] - use html at card description;
- [x] - use `enki` correct answers approach;

image

### CODE REVIEW

File: types/choose_sequence/example.json5 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    question: "Turn the arguments of a method into a array:\
+      ```function myArray() {\
+          var a = ???;\
+          a = Array.prototype.???.call(a);\
+          return a;\
+        }```",
+    answers: [
+      {
+        text: "arguments",
+        sequenceNumber: 0
```
Comment (vvscode) [inline types/choose_sequence/example.json5]: we can follow the same agreement as enki - or that's too complicated?

File: types/choose_sequence/index.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
+const getUniqueId = require('../../utils/getUniqueId');
+
+function addCodeTags(inputText) {
+  const regexp = /```(.|[^`]+)```/gs;
```
Comment (vvscode) [inline types/choose_sequence/index.js]: let's better use some md2html package (like `remark` we tried in parser)

File: types/choose_sequence/test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// @todo: move this code to common stuff
+
+require("json5/lib/register");
```
Comment (vvscode) [inline types/choose_sequence/test.js]: I will add to common script param to run tests jun on a single type

File: types/choose_sequence/json-schema.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        "answers": {
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "type": "object",
+            "required": ["text"],
+            "properties": {
+              "text": {
+                "type": "string",
```
Comment (vvscode) [inline types/choose_sequence/json-schema.json]: this one is required

File: types/choose_sequence/test.js (hunk 3 above)
Comment (vvscode) [reply types/choose_sequence/test.js]: https://github.com/memory-cards/card-types/pull/14

Comment (zenby) [reply types/choose_sequence/test.js]: At the PR description I've mentioned to delete this file after development will be complete.

File: types/choose_sequence/example.json5 (hunk 1 above)
Comment (zenby) [reply types/choose_sequence/example.json5]: You mean that for example first 2 options here will be correct answers for the question?

File: types/choose_sequence/index.js (hunk 2 above)
Comment (zenby) [reply types/choose_sequence/index.js]: Ok I'll try.

File: types/choose_sequence/test.js (hunk 3 above)
Comment (vvscode) [reply types/choose_sequence/test.js]: sure, I've just saw that for development this option might be helpfull

File: types/choose_sequence/index.js (hunk 2 above)
Comment (vvscode) [reply types/choose_sequence/index.js]: and looks like we agreed (in chat) to use html here (just sum up)

File: types/choose_sequence/example.json5 (hunk 1 above)
Comment (vvscode) [reply types/choose_sequence/example.json5]: not sure if that's clear approach - just reminded about such case

File: types/choose_sequence/json-schema.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{
+  "type": "object",
+  "required": ["type", "card", "lang", "tags"],
+  "properties": {
+    "type": {
+      "type": "string",
+      "const": "order_items"
```
Comment (zenby) [inline types/choose_sequence/json-schema.json]: `"const": "choose_sequence"`

Comment (vvscode) [reply types/choose_sequence/json-schema.json]: @zenby  вот такое валидацией должно ловиться, по идее

Comment (zenby) [reply types/choose_sequence/json-schema.json]: Оно и ловилось, только не понимал, что именно из-за этой строчки.

File: types/choose_sequence/index.js [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+            return el.sequenceNumber !== undefined;
+          })
+          .sort(function (a, b) {
+            return a.sequenceNumber - b.sequenceNumber;
+          });
+        [].forEach.call(guesses, function (el, index) {
+          el.classList.add(el.innerHTML === correctAnswers[index].text ? 'correct-guess' : 'incorrect-guess');
+        });
+
+        [].forEach.call(buttons, function (el) {
```
Comment (vvscode) [inline types/choose_sequence/index.js]: `buttons.map().join()` doesn't work?

Comment (zenby) [reply types/choose_sequence/index.js]: I tried and faced some small bugs, that's why I've decided to leave as it is.

**Pull request merged**